File: baekjoon_problems/sort/1181.py
"""
단어가 들어오면 길이가 짧은 것부터 -> 길이가 같다면 사전 순으로 정렬하라

우선 단어순으로 정렬한 후, 문자열의 길이에 따라 다시 한번 정렬을 수행한다.
"""

import sys

# 단어의 개수
N = int(sys.stdin.readline().strip())
array = set()
for _ in range(N):
    array.add(sys.stdin.readline().strip())
    
# 사전순 정렬
list_array = sorted(list(array))
# 길이 순 정렬
list_array.sort(key=lambda word: (len(word)))

for value in list_array:
    print(value)

File: baekjoon_problems/sort/1181.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val array = mutableSetOf<String>()
    repeat(n) { _ ->
        array.add(readLine())
    }

    var listArray = array.sorted()

    // 길이를 기준으로 정렬하는 Comparator
    val comparator = compareBy<String> { it.length }

    listArray = listArray.sortedWith (comparator)

    for(value in listArray) {
        println(value)
    }
}